### PR TITLE
Adds Basic Error Logging

### PR DESCRIPTION
**Pull request by kstover**
This PR utilises the "logger" system already in Ninja Forms core to add basic error logs, which are then displayed upon the system status page. This allows for support techs to see system errors without the need to load a user's site, which should save them significant time.

Currently, these are the errors captured:

- JS Errors on the front-end.
- Submission errors.
- License activation errors.

This PR also changes the way that users copy their system status information. Previously, the user clicked a button to show a textarea and then copied the text to their clipboard. This PR adds a "Copy System Status to Clipboard" button to the system status page.

How to test:

- Add a plugin or modify some JS code to get a JS error upon form display.
- Throw a PHP error upon form submission.
- Try to activate an invalid license for an add-on.

All of those actions should throw errors that are captured and displayed on the system status page. If you click the "Copy to Clipboard" button, you should be able to paste the system status elsewhere.

TODO:

This is a first-pass at error reporting for support purposes; I fully expect that this will need to be modified and evolve as support finds new ways to use it.

@wpnzach I need you to take a look at the system status page and make suggestions about wording and layout for the "Copy to Clipboard" button, as well as any instructions that you'd like the user to see in that location.

### CODE REVIEW

File: includes/AJAX/Controllers/JSError.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+class NF_AJAX_Controllers_JSError
+{
+    public function __construct()
+    {
+        add_action( 'wp_ajax_nf_log_js_error',   'ninja_forms_fucking_test'  );
+        add_action( 'wp_ajax_nopriv_nf_log_js_error',   'ninja_forms_fucking_test'  );
+    }
+
+    public function test()
```
Comment (kjohnson) [inline includes/AJAX/Controllers/JSError.php]: This seems unused, and if not used should be removed.

File: includes/AJAX/Controllers/JSError.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        var_dump( 'TEST THIS SHIT!' );
+        die();
+       
+        $this->_respond();
+    }
+
+}
+
+
+function ninja_forms_fucking_test() {
```
Comment (kjohnson) [inline includes/AJAX/Controllers/JSError.php]: This function should be a method of the class above.
Also, should probably be using a less vulgar name.

File: assets/js/admin-system-status.js [hunk 3]
```diff
@@ -18,8 +18,10 @@ jQuery.wc_strPad = function(i,l,s,w) {
     return o;
 };
 jQuery('.debug-report a').click(function(){
```
Comment (kjohnson) [inline assets/js/admin-system-status.js]: With these changes, this selector is too generic.
The "Submit a Support Request" link triggers the copy command and does not link out.

File: includes/Templates/admin-menu-system-status.html.php [hunk 4]
```diff
@@ -3,18 +3,24 @@
...
+        <div class="submit debug-report">
+    		<h3>To Get Help:</h3>
+            <ol>
+                <li><?php _e( 'Copy your System Report first with the button below' ); ?> </li>
+                <li><?php _e( 'Click "Submit a Support Request" to be directed to our site.' ); ?> </li>
+                <li><?php _e( 'Include this information in your support request by pasting in the "System Status" portion of the form. (right click, choose "Paste" or use Ctrl+V)' ); ?> </li>
+            </ol>
+            <h4><?php _e( 'This information is vital for addressing your issue in a timely manner. <em>For your security, do not post this information in public places, such as the WordPress.org support forums.</em>', 'ninja-forms' ); ?> </h4>
+            <a href="#" class="button-primary"><?php _e( 'Copy System Report', 'ninja-forms' ); ?></a>
+            <a href="https://ninjaforms.com/contact/?utm_source=plugin&utm_medium=get-help&utm_campaign=support-ticket" class="button-secondary"><?php _e( 'Submit a Support Request', 'ninja-forms' ); ?></a>
```
Comment (kjohnson) [inline includes/Templates/admin-menu-system-status.html.php]: This link out is being overridden by the `.debug-report a` selector above.

File: includes/AJAX/Controllers/JSError.php [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<?php if ( ! defined( 'ABSPATH' ) ) exit;
+
+class NF_AJAX_Controllers_JSError
+{
+    public function __construct()
+    {
+        add_action( 'wp_ajax_nf_log_js_error',   array( $this, 'log_error' );
```
Comment (kjohnson) [inline includes/AJAX/Controllers/JSError.php]: `Parse Error on includes/AJAX/Controllers/JSError.php on line 7`
Looks like the closing `)` is missing.
Same for the line below.

File: assets/js/admin-system-status.js [hunk 6]
```diff
@@ -43,11 +45,28 @@ jQuery('.debug-report a').click(function(){
             });
         }
     } );
+
     try {
-        jQuery("#debug-report").slideDown();
-        jQuery("#debug-report textarea").val( report ).focus().select();
-        paragraphContainer.slideUp();
+
+        var tmp = jQuery("<input>");
```
Comment (kjohnson) [inline assets/js/admin-system-status.js]: This needs to be a `<textarea>` for formatting purposes.

**Comment by FriendlyNGeeks** [comment]
please re add to line 104 this was missing from 3.2.1 and helps keep recaptcha rendered through multipart form
function nf_recaptcha_response( response ) {
    nfRadio.channel( 'recaptcha' ).request( 'update:response', response );
}

**Pull request merged**